src/blk/spdk/NVMEDevice.cc

```diff
@@ -90,12 +94,34 @@ class SharedDriverData {
         ns(ns_) {
     block_size = spdk_nvme_ns_get_extended_sector_size(ns);
     size = spdk_nvme_ns_get_size(ns);
+    if (trid.trtype == SPDK_NVME_TRANSPORT_PCIE) {
+      return;
+    }
+
+    if (!admin_thread.joinable()) {
+      admin_thread = std::thread(
+        [this]() {
+	  int rc;
+          while (true) {
+	    rc = spdk_nvme_ctrlr_process_admin_completions(ctrlr);
+	    if (rc < 0) {
+	      ceph_abort();
+	    }
+            sleep(1);
+          }
+        }
+      );
+    }
   }
 
   bool is_equal(const spdk_nvme_transport_id& trid2) const {
     return spdk_nvme_transport_id_compare(&trid, &trid2) == 0;
   }
   ~SharedDriverData() {
+    if (!admin_thread.joinable()) {
+      return;
+    }
+    admin_thread.join();
```

Comment:
```suggestion
    if (admin_thread.joinable()) {
      admin_thread.join();
    }
```

use positive condition for better readability.

Reply:
ack